### PR TITLE
CH4 minor cleanups

### PR DESCRIPTION
**Pull request by hajimefu**
This PR includes three commits which cleans up minor warnings/unused functions in CH4.
Please run the tests and review the changes.

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch4/all

**Comment by hajimefu** [comment]
The last patch `CH4: Remove unused probe function` became redundant due to #2462. So I'm okay with removing this patch (but keeping the other two.)

**Comment by raffenet** [comment]
@hajimefu See my alternative to your first patch and let me know what you think. I wasn't crazy about the goto usage, so I tried a more conventional approach using break. Second patch looks good and already has my signoff.

File: src/mpid/ch4/src/ch4r_proc.h [hunk 1]
```diff
@@ -87,70 +87,67 @@ MPL_STATIC_INLINE_PREFIX int MPIDIU_comm_rank_to_pid(MPIR_Comm * comm, int rank,
 #define FCNAME MPL_QUOTE(FUNCNAME)
 MPL_STATIC_INLINE_PREFIX MPIDI_av_entry_t *MPIDIU_comm_rank_to_av(MPIR_Comm * comm, int rank)
 {
-    MPIDI_av_entry_t * ret;
+    MPIDI_av_entry_t * ret = NULL;
```
Comment (hajimefu) [inline src/mpid/ch4/src/ch4r_proc.h]: I intentionally avoided this initialization which did not exist in the original code path. I have a feeling that this shouldn't be a huge performance penalty though. For the code readability I prefer your approach. Maybe @yfguo can comment.

**Comment by yfguo** [comment]
@hajimefu I will need to double check the instruction count. The initial idea of using `return` directly in the switch statement is to avoid a potential jump from the case code to a common function exit (if the function is inlined).

**Comment by hajimefu** [comment]
@yfguo another fix would be to give up using `MPIR_FUNC_VERBOSE_ENTER/EXIT` in this function (which I believe happens in several small utility functions). I don't like to see unmatched ENTER/EXIT, which is happening right now.

**Comment by yfguo** [comment]
@hajimefu @raffenet I looked the instruction counts. The use of the common exit point does not add any extra instruction (except for the printing the DBG message). I think the patch is good to go.

**Comment by raffenet** [comment]
@yfguo `break` or `goto` implementation?

**Comment by yfguo** [comment]
@raffenet The `break` implementation. `goto` is the same, but `break` is cleaner.

**Pull request merged**